packages/components/src/tooltip/test/index.js

```diff
@@ -138,27 +140,26 @@ describe( 'Tooltip', () => {
 				</Tooltip>
 			);
 
-			const button = wrapper.find( 'button' );
-			button.simulate( 'mouseenter', { type: 'mouseenter' } );
-
-			const popoverBeforeTimeout = wrapper.find( 'Popover' );
-			expect( popoverBeforeTimeout ).toHaveLength( 0 );
+			const button = screen.getByRole( 'button' );
+			await user.hover( button );
+			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
 			expect( originalMouseEnter ).toHaveBeenCalledTimes( 1 );
 
-			// Force delayedSetIsOver to be called.
 			setTimeout( () => {
-				const popoverAfterTimeout = wrapper.find( 'Popover' );
-				expect( popoverAfterTimeout ).toHaveLength( 1 );
-
+				expect( screen.getByText( 'Help text' ) ).toBeInTheDocument();
 				jest.runOnlyPendingTimers();
 				jest.useRealTimers();
 			}, TOOLTIP_DELAY );
 		} );
 
-		it( 'should respect custom delay prop when showing popover', () => {
+		it( 'should respect custom delay prop when showing popover', async () => {
```

Comment:
Let's not use `fakeTimers`, and pass two different mock function for the `onFocus` and `onMouseEnter` callbacks

```diff
diff --git a/packages/components/src/tooltip/test/index.js b/packages/components/src/tooltip/test/index.js
index 3702188102..1057d6eee1 100644
--- a/packages/components/src/tooltip/test/index.js
+++ b/packages/components/src/tooltip/test/index.js
@@ -123,41 +123,48 @@ describe( 'Tooltip', () => {
 			);
 		} );
 
-		it( 'should respect custom delay prop when showing popover', async () => {
+		it( 'should respect custom delay prop when showing tooltip', async () => {
 			const user = userEvent.setup( {
 				advanceTimers: jest.advanceTimersByTime,
 			} );
 
-			const originalMouseEnter = jest.fn();
-			jest.useFakeTimers();
+			const TEST_DELAY = TOOLTIP_DELAY * 2;
+
+			const mockOnMouseEnter = jest.fn();
+			const mockOnFocus = jest.fn();
 			render(
-				<Tooltip text="Help text" delay={ 2000 }>
+				<Tooltip text="Help text" delay={ TEST_DELAY }>
 					<button
-						onMouseEnter={ originalMouseEnter }
-						onFocus={ originalMouseEnter }
+						onMouseEnter={ mockOnMouseEnter }
+						onFocus={ mockOnFocus }
 					>
 						<span>Hover Me!</span>
 					</button>
 				</Tooltip>
 			);
 
-			const button = screen.getByRole( 'button' );
+			const button = screen.getByRole( 'button', { name: 'Hover Me!' } );
+			expect( button ).toBeInTheDocument();
+
 			await user.hover( button );
+
+			// Tooltip hasn't appeared yet
 			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
-			expect( originalMouseEnter ).toHaveBeenCalledTimes( 1 );
+			expect( mockOnMouseEnter ).toHaveBeenCalledTimes( 1 );
 
-			// Popover does not yet exist after default delay, because custom delay is passed.
-			setTimeout( () => {
-				expect(
-					screen.queryByText( 'Help text' )
-				).not.toBeInTheDocument();
-			}, TOOLTIP_DELAY );
-			// Popover appears after custom delay.
-			setTimeout( () => {
-				expect( screen.getByText( 'Help text' ) ).toBeInTheDocument();
-				jest.runOnlyPendingTimers();
-				jest.useRealTimers();
-			}, 2000 );
+			// Advance by the usual TOOLTIP_DELAY
+			act( () => jest.advanceTimersByTime( TOOLTIP_DELAY ) );
+
+			// Tooltip hasn't appeared yet after the usual delay
+			expect( screen.queryByText( 'Help text' ) ).not.toBeInTheDocument();
+
+			// Advance time again, so that we reach the full TEST_DELAY time
+			act( () => jest.advanceTimersByTime( TEST_DELAY - TOOLTIP_DELAY ) );
+
+			// Tooltip shows after TEST_DELAY time
+			expect( screen.getByText( 'Help text' ) ).toBeInTheDocument();
+
+			expect( mockOnFocus ).not.toHaveBeenCalled();
 		} );
 
 		it( 'should show tooltip when an element is disabled', async () => {
```

Reply:
Fixed by 4b143a0